### PR TITLE
Update rules.md, options.md and plugins.md

### PR DESCRIPTION
**Pull request by Mouvedia**
> Which issue, if any, is this issue related to?

#2643

> Is there anything in the PR that needs further explanation?

blocks #7895

### CODE REVIEW

File: docs/user-guide/options.md [hunk 1]
```diff
@@ -40,12 +40,12 @@ For CSS with standard syntax, Stylelint uses [postcss-safe-parser](https://www.n
 
 When using the Node.js API, the autofixed code is available as the value of the `code` property in the returned object.
 
-If a source contains a:
+---
 
-- scoped disable comment, e.g. `/* stylelint-disable color-named */`, any problems reported by the scoped rules will not be automatically fixed anywhere in the source
-- unscoped disable comment, i.e. `/* stylelint-disable */`, the entirety of source will not be automatically fixed
+When a rule relies on the deprecated `context.fix` and a source contains:
```
Comment (ybiquitous) [inline docs/user-guide/options.md:45]: [suggest] Minor. Making `context.fix` a link may be nice for people who want to know the deprecation. For example:

```suggestion
When a rule relies on the deprecated [`context.fix`](../developer-guide/rules.md#context) and a source contains:
```

File: docs/developer-guide/plugins.md [hunk 2]
```diff
@@ -99,7 +99,7 @@ For your plugin rule to work with the [standard configuration format](../user-gu
 - the primary option
 - optionally, a secondary options object
 
-If your plugin rule supports [autofixing](rules.md#add-autofix), then `ruleFunction` should also accept a third argument: `context`.
+If your plugin rule supports [autofixing](rules.md#add-autofix), then `fixable` should be set to `true` in `meta`.
```
Comment (ybiquitous) [inline docs/developer-guide/plugins.md:102]: [suggest] How about this change for consistency?

https://github.com/stylelint/stylelint/blob/066fc8d8c9a3bf6002b9d13e2f85d9db99cf8f97/docs/developer-guide/rules.md?plain=1#L192

```suggestion
If your plugin rule supports [autofixing](rules.md#add-autofix), then `meta.fixable` should be set to `true`.
```

File: docs/user-guide/options.md [hunk 3]
```diff
@@ -40,12 +40,12 @@ For CSS with standard syntax, Stylelint uses [postcss-safe-parser](https://www.n
 
 When using the Node.js API, the autofixed code is available as the value of the `code` property in the returned object.
 
-If a source contains a:
+---
```
Comment (ybiquitous) [inline docs/user-guide/options.md:43]: [suggest] Minor. Personally, I feel this bar is a bit verbose. Each heading has an underline, so I am concerned that people may be a bit confused about which section is separated.

```suggestion
```

**Review by ybiquitous — COMMENTED** [review]
@Mouvedia Thanks for the pull request. Overall LGTM, but I left some suggestions. Can you please take a look?

File: docs/developer-guide/rules.md [hunk 4]
```diff
@@ -198,23 +198,37 @@ const meta = {
...
+### Context
+
 `context` is an object which could have three properties:
 
 - `configurationComment`(string): String that prefixes configuration comments like `/* stylelint-disable */`.
 - `fix`(boolean): If `true`, your rule can apply autofixes.
 - `newline`(string): Line-ending used in current linted file.
 
+> [!WARNING]
+> The `fix` boolean is deprecated in favour of the `fix` callback which properly handles [configuration comments](../user-guide/ignore-code.md#parts-of-a-file).
```
[the remark this answers is not in the document]
Comment (Mouvedia) [reply docs/developer-guide/rules.md:230]:
```suggestion
> The convention of restricting the appliance of fixes based on the `context.fix` property is deprecated in favour of recommending the `fix` callback which properly handles [configuration comments](../user-guide/ignore-code.md#parts-of-a-file).
```

File: docs/user-guide/options.md (hunk 1 above)
Comment (Mouvedia) [reply docs/user-guide/options.md:45]:
```suggestion
When a rule relies on the deprecated [`context`](../developer-guide/rules.md#context).fix` and a source contains:
```

File: docs/user-guide/options.md [hunk 5]
```diff
@@ -40,12 +40,12 @@ For CSS with standard syntax, Stylelint uses [postcss-safe-parser](https://www.n
...
```
Comment (Mouvedia) [inline docs/user-guide/options.md:44]:
```suggestion
```

File: docs/user-guide/options.md (hunk 3 above)
Comment (Mouvedia) [inline docs/user-guide/options.md:43]:
```suggestion
```

**Review by ybiquitous — APPROVED** [review]
Thanks. Lgtm 👍

**Pull request merged**